src/main/java/org/spongepowered/api/plugin/PluginContainer.java

```diff
@@ -52,10 +57,19 @@
     String getVersion();
 
     /**
-     * Returns the created instance of {@link Plugin}.
+     * Returns the assigned logger to this {@link Plugin}.
+     *
+     * @return The assigned logger
+     */
+    default Logger getLogger() {
+        return LoggerFactory.getLogger(getId());
+    }
+
+    /**
+     * Returns the created instance of {@link Plugin} if it is available.
      *
-     * @return The instance
+     * @return The instance if available
      */
-    Object getInstance();
+    Optional<Object> getInstance();
```

Comment:
How about returning dummies if we need to for things like the MCP container, and throwing an exception if we for some reason have a mod from forge which has no instance attached, since I think that case does qualify as `exceptional` eh?